client-server/src/main/java/SerializableStrategy.java

```diff
@@ -0,0 +1,20 @@
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+public class SerializableStrategy implements MessageStrategy {
+    @Override
+    public void send(ObjectOutputStream oos, Message msg) throws IOException {
+        oos.writeObject(msg);
+        oos.flush();
+    }
+
+    @Override
+    public Message receive(ObjectInputStream ois) throws IOException {
+        try {
+            return (Message) ois.readObject();
+        } catch (ClassNotFoundException e) {
+            throw new IllegalStateException(e);
```

Comment:
What will happen if this line is called (new client has sent different Message or so)?

Reply:
done ??

Reply:
Nope, you still will stop the client, once it receives unexpected class.
It's ok to throw `IllegalStateException(e)` here, but think where you catch this (IllegalStateException) exception after it?